docs/reference/runtime-adapter.md

```diff
@@ -0,0 +1,251 @@
+# Using `RuntimeAdapter` in userland
+
+Any package generated through `universal-middleware` exposes a `RuntimeAdapter` type.
+
+It can be used to type `runtime` when exposed to userland.
+
+### Adapter specific properties
+
+::: code-group
+
+```ts twoslash [hono]
+// @noErrors
+import type { Runtime, HonoAdapter } from "@universal-middleware/core";
+
+export type Explain<A extends any> =
+  A extends Function
+    ? A
+    : {[K in keyof A]: A[K]} & unknown
+
+type RuntimeAdapter = Explain<HonoAdapter>;
+
+// ---cut---
+import type { RuntimeAdapter } from "somelib/hono";
+
+const runtime: RuntimeAdapter;
+
+// original Hono Context. See https://hono.dev/docs/api/context
+runtime.hono;
+//      ^^^^
+```
+
+```ts twoslash [h3]
```

Comment:
Would be nice if the tab elements appears in order of popularity / what we like most. E.g. `express` first, then `hono`, then cloudflare, etc.

Reply:
(Or `hono` first we feel like it.)

Reply:
https://github.com/magne4000/universal-middleware/commit/58196240017cf75472cadf65017128643f53c151